lib/AST/ASTContext.cpp

```diff
@@ -498,6 +498,51 @@ void ASTContext::lookupInSwiftModule(
   M->lookupValue({ }, identifier, NLKind::UnqualifiedLookup, results);
 }
 
+FuncDecl *ASTContext::lookupArrayAppendElementDecl() {
+  if (ArrayAppendElementDecl)
+    return ArrayAppendElementDecl;
+
+  auto AppendFunctions = getArrayDecl()->lookupDirect(getIdentifier("append"));
+
+  for (auto CandidateFn : AppendFunctions) {
+    auto FnDecl = dyn_cast<FuncDecl>(CandidateFn);
+    auto Attrs = FnDecl->getAttrs();
+    for (auto *A : Attrs.getAttributes<SemanticsAttr, false>()) {
+      if (A->Value == "array.append_element") {
+        auto ParamLists = FnDecl->getParameterLists();
+        assert(ParamLists.size() == 2 && "Should have two parameter lists");
```

Comment:
This whole set of asserts should be better put into an `#ifndef NDEBUG` block. See ASTContext.cpp for examples.